# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/xf86-video-xgi/fix-hwcursor-64bit.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- xf86-video-xgi/src/xgi_accel.c.vanilla	2024-04-04 11:00:00.022794539 +0000
+++ xf86-video-xgi/src/xgi_accel.c	2024-04-04 11:01:27.518797785 +0000
@@ -60,6 +60,10 @@
 
 #define DEV_HEIGHT	0xfff	/* "Device height of destination bitmap" */
 
+int Alignment;
+unsigned long    r_port, w_port ;
+int      G2CmdQueLen;
+
 /* Jong 07/24/2008; performance tuning */
 long	g_srcbase=-1;
 long	g_dstbase=-1;
--- xf86-video-xgi/src/xgi_regs.h.vanilla	2024-04-04 11:36:43.445876287 +0000
+++ xf86-video-xgi/src/xgi_regs.h	2024-04-04 11:37:21.448877697 +0000
@@ -232,7 +232,7 @@
 #define GPIOH_WRITE (1<<7)
 #define GPIOH_READ  (1<<0)
 
-#define XGIMMIOLONG(offset)  *(volatile unsigned long *)(pXGI->IOBase+(offset))
+#define XGIMMIOLONG(offset)  *(volatile unsigned int *)(pXGI->IOBase+(offset))
 #define XGIMMIOSHORT(offset) *(volatile unsigned short *)(pXGI->IOBase+(offset))
 #define XGIMMIOBYTE(offset)  *(volatile unsigned char *)(pXGI->IOBase+(offset))
 
--- xf86-video-xgi/src/xgi_cursor.c.vanilla	2024-04-04 12:07:14.077944204 +0000
+++ xf86-video-xgi/src/xgi_cursor.c	2024-04-04 12:07:11.422944106 +0000
@@ -162,6 +162,8 @@
     unsigned char *const pCursorShape =
         pXGI->FbBase + pXGI->CursorOffset;
 
+    pXGI->HWARGBCursor = FALSE;
+
     memcpy(pCursorShape, src, 1024);
 
     xgiG2CRT1_SetCursorAddressPattern(cursor_base,0) ;
